### PR TITLE
fix: address iframe resizing issue in jupyter notebooks

### PR DESCRIPTION
**Pull request by dakshpokar**
## Description
This PR includes changes for removing overflow on iframes and resizing them according to content of the plot.

## Type of Change

- [x] Bug fix
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules

# Pull Request

## Description

This PR enhances iframe resizing behavior to dynamically adjust height and width based on content size and user interaction. Specifically, the height is recalculated with additional space (100px) when the braille-input field is focused, and 50px otherwise. This ensures the iframe is not clipped during braille mode. The script also adds event listeners for focusin, focusout, and resize events to trigger appropriate resizing as needed, which were added by @krishnanand5.

### CODE REVIEW

File: maidr/core/maidr.py [hunk 1]
```diff
@@ -193,24 +193,36 @@ def generate_iframe_script(unique_id: str) -> str:
             resizing_script = f"""
                 function resizeIframe() {{
                     let iframe = document.getElementById('{unique_id}');
+
                     if (iframe && iframe.contentWindow && iframe.contentWindow.document) {{
                         let iframeDocument = iframe.contentWindow.document;
-                        let brailleContainer = iframeDocument.getElementById('braille-div');
+                        let brailleContainer = iframeDocument.getElementById('braille-input');
+
+                        iframe.style.height = 'auto';
```
Comment (krishnanand5) [inline maidr/core/maidr.py:201]: Greetings @dakshpokar,

I think we can set the height to auto in the `tags.iframe` construction. We set the height to 100% there and we are overriding it here. I removed this assignment and checked on Shiny by setting height to auto in `tags.iframe` construction - seems to work fine. Do check with Jupyter and share your thoughts on this.

File: maidr/core/maidr.py [hunk 2]
```diff
@@ -193,24 +193,36 @@ def generate_iframe_script(unique_id: str) -> str:
...
-                    iframe.contentWindow.document.addEventListener('focusout', resizeIframe);
                 }};
+                iframe.contentWindow.document.addEventListener('focusin', () => {{
+                    resizeIframe();
+                }});
+                iframe.contentWindow.document.addEventListener('focusout', () => {{
+                    resizeIframe();
+                }});
             """
             return resizing_script
```
Comment (krishnanand5) [inline maidr/core/maidr.py:228]: on the `tags.iframe` assignment, I did check your old PR where you have loaded the LLM credentials from environment storage. My PR from yesterday inadvertently removed that assignment. You can refer to your old PR to add that to the `onLoad` argument in `tags.iframe`.

**Review by krishnanand5 — COMMENTED** [review]
Hey @dakshpokar,

The changes are fine on Shiny. Thank you for taking care of the dynamic height adjustment when braille is active. Let me know if we can address my comments.

Regards,
Krishna Anandan Ganesan.

File: maidr/core/maidr.py (hunk 2 above)
Comment (dakshpokar) [reply maidr/core/maidr.py:228]: @krishnanand5 We had to remove the LLM keys injection script, due a security flaw that we discovered in #119.

File: maidr/core/maidr.py (hunk 1 above)
Comment (dakshpokar) [reply maidr/core/maidr.py:201]: That's true I was going to add in the tags but the problem with that is that everytime the focus in and out script runs I want the original scrollable height of the content inside iframe. If we don't reset the previous height, the script was adding additional (not required) height. Hence, I just reset the height when the script runs so as to get the minimum scrollable height and then on the basis of that do the rest of the calculations.

Comment (krishnanand5) [reply maidr/core/maidr.py:201]: Sure, in that case this change is feasible.

**Review by krishnanand5 — APPROVED** [review]

**Review by jooyoungseo — APPROVED** [review]
LGTM!

**Pull request merged**